src/content_negotiation/language_negotiation.py

```diff
@@ -0,0 +1,144 @@
+"""Module for determining content-language based on accept-language header.
+
+Example:
+    >>> from content_negotiation import decide_language, NoAgreeableLanguageError
+    >>>
+    >>> accept_language_headers = ["en-GB;q=0.8", "nb-NO;q=0.9"]
+    >>> supported_languages = ["en-GB", "en", "nb-NO", "nb", "en-US"]
+    >>>
+    >>> try:
+    >>>     content_language = decide_language(accept_language_headers, supported_languages)
+    >>> except NoAgreeableLanguageError:
+    >>>     print("No agreeable language found.")
+    >>>     # Handle error, by returning e.g. 406 Not Acceptable
+    >>> print(content_language)
+    'nb-NO'
+"""
+import logging
+from typing import Any, List
+
+
+class NoAgreeableLanguageError(Exception):
+    """Exception for no agreeable language."""
+
+    pass
+
+
+class WeightedLanguage:
+    """Class for handling weighted languages."""
+
+    language: str
+    q: float = 1.0
+
+    def __init__(self, language: str) -> None:
+        """Initialize the weighted language."""
+        weighted_language_split = language.split(";")
+        # Instantiate weighted language:
+        logging.debug(f"Assigning q-parameter for weighted languag: {language}")
+        self.language = weighted_language_split[0]
+
+        # If q-parameter is present, assign it:
+        for weighted_language_part in weighted_language_split[1:]:
+            if weighted_language_part.startswith("q="):
+                self.q = float(
+                    # RFC specifies only 3 decimals may be used in q value.
+                    weighted_language_part.split("=")[1][0:5]
+                )
+
+    def __eq__(self, other: Any) -> bool:
+        """Compare two weighted languages."""
+        if isinstance(other, str):
+            return self.language == other
+        return False  # pragma: no cover
+
+    def __lt__(self, other: Any) -> bool:
+        """Compare two weighted languages."""
+        if isinstance(other, WeightedLanguage):
+            # If weighted languages are not equal, compare q value:
```

Comment:
Forstår ikke helt hva du prøver å si her
```suggestion
            # If weighted languages are not equal, compare q value:
```